docs/getting-started-guides/mesos.md

```diff
@@ -42,65 +45,72 @@ The internal IP address of the master may be obtained via `hostname -i`.
 
 ```bash
 $ export servicehost=$(hostname -i)
-$ export mesos_master=${servicehost}:5050
 $ export KUBERNETES_MASTER=http://${servicehost}:8888
 ```
+
 ### Deploy etcd
 Start etcd and verify that it is running:
 
 ```bash
-$ sudo docker run -d --hostname $(uname -n) --name etcd -p 4001:4001 -p 7001:7001 coreos/etcd
+$ sudo docker run -d --hostname $(uname -n) --name etcd -p 4001:4001 -p 7001:7001 quay.io/coreos/etcd:v2.0.12
 ```
 
 ```bash
 $ sudo docker ps
-CONTAINER ID   IMAGE                COMMAND   CREATED   STATUS   PORTS                NAMES
-fd7bac9e2301   coreos/etcd:latest   "/etcd"   5s ago    Up 3s    2379/tcp, 2380/...   etcd
+CONTAINER ID   IMAGE                        COMMAND   CREATED   STATUS   PORTS                NAMES
+fd7bac9e2301   quay.io/coreos/etcd:v2.0.12  "/etcd"   5s ago    Up 3s    2379/tcp, 2380/...   etcd
 ```
 It's also a good idea to ensure your etcd instance is reachable by testing it
 ```bash
-curl -L http://$servicehost:4001/v2/keys/
+curl -L http://${servicehost}:4001/v2/keys/
 ```
 If connectivity is OK, you will see an output of the available keys in etcd (if any).
 
 ### Start Kubernetes-Mesos Services
-Start the kubernetes-mesos API server, controller manager, and scheduler on a Mesos master node:
+Update your PATH to more easily run the Kubernetes-Mesos binaries:
+```bash
+$ export PATH="$(pwd)/_output/local/go/bin:$PATH"
+```
+Identify your Mesos master: depending on your Mesos installation this is either a `host:port` like `${servicehost}:5050` or a ZooKeeper URL like `zk://${servicehost}:2181/mesos`.
+In order to let Kubernetes survive Mesos master changes, the ZooKeeper URL is recommended for production environments.
+```bash
+$ export MESOS_MASTER=<host:port or zk:// url>
+```
+Create a cloud config file `mesos-cloud.conf` in the current directory with the following contents:
+```bash
+$ cat <<EOF >mesos-cloud.conf
+[mesos-cloud]
+        mesos-master        = ${MESOS_MASTER}
+EOF
+```
+
+Now start the kubernetes-mesos API server, controller manager, and scheduler on a Mesos master node:
```

Comment:
should this be "the Mesos master node" ? I know you can do HA masters so in that situation there can be multiple, but if the user follows the default mesos setup instructions do they get that?